src/vanilla/shallow.ts

```diff
@@ -49,6 +52,9 @@ export function shallow<T>(valueA: T, valueB: T): boolean {
   if (Object.is(valueA, valueB)) {
     return true
   }
+  if (type(valueA) !== type(valueB)) {
+    return false
+  }
```

Comment:
Clever! I didn't know about the `Object.prototype.toString` trick. I think the block below be removed

```
if (
    typeof valueA !== 'object' ||
    valueA === null ||
    typeof valueB !== 'object' ||
    valueB === null
  ) {
    return false
  }
  ```

Reply:
actually, we need it

Reply:
Oh I see what it's doing now. Sounds good